pkg/networkmanager/firewall.jsx

```diff
@@ -374,6 +376,15 @@ class AddServicesModal extends React.Component {
         });
     }
 
+    onToggleZone(event) {
+        let zone = event.target.value;
+        this.setState(state => {
+            if (state.zones.indexOf(zone) === -1)
+                return { zones: state.zones.concat(zone) };
+            return { zones: state.zones.filter(z => z !== zone) };
+        });
```

Comment:
Might be easier to make `zones` a Set.

Reply:
I tried this but it turned out to be more verbose with copying it first to modify it etc... 

Plus this exact logic is used in another place of the firewall.